File: spectra-starter/spectra-starter-common/src/main/java/com/yangxj96/spectra/starter/common/converter/StringToDateConverter.java
package com.yangxj96.spectra.starter.common.converter;

import cn.hutool.core.date.DateUtil;
import com.yangxj96.spectra.starter.common.props.JacksonProperties;
import jakarta.annotation.Resource;
import org.springframework.core.convert.converter.Converter;
import org.springframework.lang.Nullable;

import java.util.Date;

/**
 * 字符串转LocalDatetime的Converter
 */
public class StringToDateConverter implements Converter<String, Date> {

    @Resource
    private JacksonProperties props;

    @Override
    @Nullable
    public Date convert(String source) {
        return DateUtil.parse(source.trim(), props.getLocalDateTimeFormat());
    }
}

File: yangxj96-starter/yangxj96-starter-common/src/main/kotlin/com/yangxj96/saas/starter/common/converter/StringToDateConverter.kt
package com.yangxj96.saas.starter.common.converter

import cn.hutool.core.date.DateUtil
import com.yangxj96.saas.starter.common.props.JacksonProperties
import jakarta.annotation.Resource
import org.springframework.context.annotation.Configuration
import org.springframework.core.convert.converter.Converter
import java.util.*


/**
 * 字符串转LocalDatetime的Converter
 */
class StringToDateConverter : Converter<String, Date> {

    @Resource
    private lateinit var props: JacksonProperties

    override fun convert(source: String): Date? {
        return DateUtil.parse(source.trim { it <= ' ' }, props.localDateTimeFormat)
    }

}